Lib/http/server.py

```diff
@@ -687,25 +687,30 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
+        fs = os.fstat(f.fileno())
+            
         try:
             # Use browser cache if possible
-            if "If-Modified-Since" in self.headers:
+            if "If-Modified-Since" in self.headers \
+                    and not "If-None-Match" in self.headers:
                 # compare If-Modified-Since and date of last file modification
-                fs = os.stat(path)
                 ims = email.utils.parsedate(self.headers["If-Modified-Since"])
```

Comment:
It looks a problem to me that `email.utils.parsedate()` silently ignores timezone. Maybe use `email.utils.parsedate_to_datetime()` instead?